src/Rec/Rec.ts

```diff
@@ -0,0 +1,467 @@
+    ) =>
+    <K extends string>(rec: Rec<K, V>): readonly V[] => {
+        const values: V[] = []
+
+        for (const k in rec) {
+            values.push(rec[k])
+        }
+
+        return values.sort(orderingComparer.compare)
+    }
+
+/**
+ * Returns the record as an array of key-value tuples. The array will be sorted by
+ * key, using the given `OrderingComparer` or falling back to the default ASCII-based
+ * sort.
+ *
+ * @group Transformations
+ * @group Utils
+ */
+export const toArray =
+    <K extends string>(orderingComparer?: OrderingComparer<K>) =>
+    <V>(rec: Rec<K, V>): readonly (readonly [K, V])[] =>
+        pipe(rec, keys(orderingComparer), _ => _.map(k => [k, rec[k]]))
+
+/**
+ * Also commonly referred to as `fold` or `aggregate`. Applies each key/value
+ * pair in the map to a "reducer" (or "folding") function to build up a final
+ * accumulated value.
+ *
+ * Key/value pairs will be given to the reducer function based on the sort-order
+ * of the _keys_. That order can be specified by passing the `OrderingComparer`.
+ * Defaults to the standard ASCII-based sort.
+ *
+ * @group Transformations
+ * @group Utils
+ *
+ * @param f
+ * The reducer function. Accepts the accumulator value, the current key, and the
+ * current value, and must return the next (incremental) accumulator value.
+ */
+export const reduce =
+    <S, K extends string, V>(
+        init: S,
+        f: (acc: S, k: NoInfer<K>, v: NoInfer<V>) => S,
+        orderingComparer?: OrderingComparer<NoInfer<K>>
+    ) =>
+    (rec: Rec<K, V>): S =>
+        pipe(rec, toArray(orderingComparer), _ =>
+            _.reduce((s, [k, v]) => f(s, k, v), init)
+        )
+
+/**
+ * Like {@link reduce}, but the key-value pairs are passed to the reducer in
+ * _reverse_ sort-order.
+ */
+export const reduceRight =
+    <S, K extends string, V>(
+        init: S,
+        f: (acc: S, k: NoInfer<K>, v: NoInfer<V>) => S,
+        orderingComparer?: OrderingComparer<NoInfer<K>>
+    ) =>
+    (rec: Rec<K, V>): S =>
+        pipe(rec, toArray(orderingComparer), _ =>
+            _.reduceRight((s, [k, v]) => f(s, k, v), init)
+        )
+/**
+ * Get a new record containing only the key/value pairs for which the given
+ * predicate function returns `true`.
+ *
+ * @group Transformations
+ * @group Filtering
+ */
+export const filter =
+    <K extends string, V>(f: (k: NoInfer<K>, v: NoInfer<V>) => boolean) =>
+    (rec: Rec<K, V>): Rec<K, V> => {
+        if (isEmpty(rec)) {
+            return rec
+        }
+
+        const out: Record<K, V> = empty<K, V>()
+
+        for (const k in rec) {
+            if (f(k, rec[k])) {
+                out[k] = rec[k]
+            }
+        }
+
+        return out
+    }
+
+/**
+ * Test whether every key/value pair in a record returns `true` for the
+ * given predicate function.
+ *
+ * @group Utils
+ */
+export const every =
+    <K extends string, V>(f: (k: NoInfer<K>, v: NoInfer<V>) => boolean) =>
+    (rec: Rec<K, V>): boolean => {
```

Comment:
⛏️ / 🎨 Could we just do `toArray(rec).every(f)` ? aside: I noticed we don't have `Array.every` (what I originally reached for here)

Reply:
In this case, I think I prefer the explicit loop because `toArray` + `every` requires two iterations over the object instead of just one